token/cli/src/config.rs

```diff
@@ -282,7 +288,10 @@ impl<'a> Config<'a> {
 
         let nonce_blockhash = value_of(matches, BLOCKHASH_ARG.name);
         let compute_unit_price = value_of(matches, COMPUTE_UNIT_PRICE_ARG.name);
-        let compute_unit_limit = value_of(matches, COMPUTE_UNIT_LIMIT_ARG.name);
+        let compute_unit_limit = matches
+            .value_of(COMPUTE_UNIT_LIMIT_ARG.name)
+            .map(|x| parse_compute_unit_limit(x).unwrap())
+            .unwrap_or(ComputeUnitLimit::Default);
```

Comment:
So this is my thought for how to handle without a special "SIMULATED" input:

```suggestion
        let compute_unit_limit = matches
            .value_of(COMPUTE_UNIT_LIMIT_ARG.name);
        if compute_unit_limit.is_none() && compute_unit_price.is_some() && (sign_only || nonce_blockhash.is_some()) {
            <return clap error that compute_unit_limit must be specified> // Possibly this if case could be elimintated by using clap `requires()` configs in the App
        }
        let compute_unit_limit = compute_unit_limit
            .map(|x| parse_compute_unit_limit(x).unwrap())
            .unwrap_or_else(|| if sign_only || nonce_blockhash.is_some() { ComputeUnitLimit::Default } else { ComputeUnitLimit::Simulate });
```
(Sorry about the poor formatting.)
Wdyt?

Reply:
That works for me! Just so I'm sure I understand, this still allows for `SIMULATED` as an option, but is smarter about when to default to simulating transactions, correct?

What's the rationale for checking `sign_only`? Is the idea to support offline signing *without* a nonce? That will still fail to broadcast since it'll try to simulate during the actual send by default.

Reply:
> this still allows for `SIMULATED` as an option, but is smarter about when to default to simulating transactions, correct?

No, my thinking was to eliminate the `SIMULATED` input option, but keep the `ComputeUnitLimit::Simulate` variant to use for the "right" transactions.

> What's the rationale for checking sign_only? Is the idea to support offline signing without a nonce? That will still fail to broadcast since it'll try to simulate during the actual send by default.

The rationale was to prevent simulation for both the offline steps, ie: (1) signing command, which has `--sign-only`; and (2) the send-tx-with-presigner command, which must have `--blockhash` ([link](https://github.com/anza-xyz/agave/blob/4672efa1c1cedcf0f034a62b8b72fb1aa0603e3a/clap-utils/src/offline.rs#L53)).
But looking at the clap-utils code again, I see checking `--blockhash` should be sufficient for both cases, since `--sign-only` does require `--blockhash` as well. Sorry about making that suggestion more complicated than it needs to be.

Reply:
Thanks for the clarification, works for me!